pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -284,6 +284,15 @@ impl InMemoryLayer {
     ) -> Result<()> {
         trace!("put_value key {} at {}/{}", key, self.timeline_id, lsn);
 
+        if let Value::Image(buf) = val {
+            if buf.len() > MAX_VALUE_SIZE {
+                tracing::warn!(
+                    "Can't put value of size {} above limit {MAX_VALUE_SIZE} for key {key}",
+                    buf.len()
+                );
+            }
+        }
+
```

Comment:
Bad thing about this is that there would be a lot of warnings if such code went into prod.

Reply:
> Bad thing about this is that there would be a lot of warnings if such code went into prod.

If there is instances where we have such large values, it is good to know about them. If there is too many, we should increase the threshold.

> I thought the problem is that we fail to stop appending when we've gone over the size target because we simply do not check for it.

Right now that is the problem, yes. But if a value is 100 MB large, it will cause us to create delta layers larger than our target size, as even if you just put that value there on its own, it alone will cause that bloat.

If you have a value that is 5 GB large, then we can't upload that delta layer any more.

Reply:
I still hold that we should not be logging these in prod -- in tests (`#[cfg(feature = "testing")]`) it would make sense as it would fail the test case. If we had had this in prod, we would be dangering a self-DOS and losing of quite vital for debugging information. This line would had printed some 500 times per layer flush.

I think this kind of testing has a sort of overlap with the actual fix to stop appending too many records to a layer file: it is unknown if we can performantly enough serialize the Value to count the amount of bytes it creates -- most likely yes.

Reply:
As discussed there were no over 10MB values 500 times per layer flush.